src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.Encrypt.cs

```diff
@@ -266,8 +266,49 @@ private static CMSG_RECIPIENT_ENCODE_INFO EncodeRecipientInfo(CmsRecipient recip
 
                     pEncodeInfo->cbSize = sizeof(CMSG_KEY_TRANS_RECIPIENT_ENCODE_INFO);
 
-                    CRYPT_ALGORITHM_IDENTIFIER algId = pCertInfo->SubjectPublicKeyInfo.Algorithm;
-                    pEncodeInfo->KeyEncryptionAlgorithm = algId;
+                    if (recipient.RSAEncryptionPadding is null)
+                    {
+                        CRYPT_ALGORITHM_IDENTIFIER algId = pCertInfo->SubjectPublicKeyInfo.Algorithm;
+                        pEncodeInfo->KeyEncryptionAlgorithm = algId;
+                    }
+                    else if (recipient.RSAEncryptionPadding.Mode == RSAEncryptionPaddingMode.Pkcs1)
+                    {
+                        pEncodeInfo->KeyEncryptionAlgorithm.pszObjId = hb.AllocAsciiString(Oids.Rsa);
+                        pEncodeInfo->KeyEncryptionAlgorithm.Parameters.cbData = (uint)s_rsaPkcsParameters.Length;
+                        pEncodeInfo->KeyEncryptionAlgorithm.Parameters.pbData = hb.AllocBytes(s_rsaPkcsParameters);
+                    }
+                    else if (recipient.RSAEncryptionPadding.Mode == RSAEncryptionPaddingMode.Oaep &&
+                             recipient.RSAEncryptionPadding.OaepHashAlgorithm == HashAlgorithmName.SHA1)
```

Comment:
Why not `else if (recipient.RSAEncryptionPadding == RSAEncryptionPadding.OaepSha1)` ?

Reply:
Oh, I suppose I goofed looking at `RSAEncryptionPadding` docs - I didn't see that it implemented equality.

Reply:
No worries, I had to look it up, too :smile: